### PR TITLE
Copy AL proxy shape internal data on duplication.

### PR DESCRIPTION
**Pull request by ppt-adsk**

### CODE REVIEW

File: lib/ufe/ProxyShapeHandler.cpp [hunk 1]
```diff
@@ -61,12 +61,8 @@ std::vector<std::string> ProxyShapeHandler::getAllNames()
 /*static*/
 UsdStageWeakPtr ProxyShapeHandler::dagPathToStage(const std::string& dagPath)
 {
-    UsdStageWeakPtr stagePtr;
-
-    if (UsdPrim usdPrim = UsdMayaQuery::GetPrim(dagPath))
-        stagePtr = usdPrim.GetStage();
-
-    return stagePtr;
+	auto prim = UsdMayaQuery::GetPrim(dagPath);
```
Comment (ppt-adsk) [inline lib/ufe/ProxyShapeHandler.cpp]: No change in behavior, was just in my commit, so I kept it.

File: plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h [hunk 2]
```diff
@@ -1005,6 +1005,7 @@ class ProxyShape
   MStatus preEvaluation(const MDGContext & context, const MEvaluationNode& evaluationNode) override;
   void CacheEmptyBoundingBox(MBoundingBox&) override;
   UsdTimeCode GetOutputTime(MDataBlock) const override;
+  void copyInternalData(MPxNode* srcNode) override;
```
Comment (ppt-adsk) [inline plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h]: Missing call to hook for duplicating Maya node private data.

**Comment by kxl-adsk** [comment]
@cfmoore007 this is a complete fix for #69 , i.e. including a proper way to duplicate proxy shape private data.

**Comment by cfmoore007** [comment]
@kxl-adsk  - excellent, will try it out now, and update the other issue accordingly.

**Comment by cfmoore007** [comment]
1) Confirmed, this works ( compiled this PR against latest dev )  :

duplicateNoCrash

No crash.

2) Edits to the original stage also are preserved when duplicated :

duplicateEditsPreserved

3) However, there's a bug, where if you move the sphere ( I will log this separately ) :
duplicateMovesSameTime

**Comment by ppt-adsk** [comment]
Unit test added.  Without the fix, test fails, with the fix, test passes.

**Pull request merged**